### PR TITLE
upcoming: [DI-21119] - Handle jwe token limit in ACLP UI and Bug fix for resources component

### PR DESCRIPTION
**Pull request by ankitaakamai**
## Description 📝

 Handled jwe token limit in ACLP UI and Bug fix for resources component.

## Changes  🔄

List any change(s) relevant to the reviewer.

- JWE token will only be created for selected resources/database clusters. It was earlier created for all the available resources of selected service type.
- Resources/Database clusters autocomplete has a bug fix - 
     When these three are equal => 1. available resources/clusters 2. selected resources/clusters 3. max resource selection limit, select/deselect all will work as expected.  Earlier, in this edge case, deselect all got disabled after maximum resources selection. 

## Target release date 🗓️

2-Dec-2024

## Preview 📷

| Before  | After   |
| ------- | ------- | 
| image | image |

### Verification steps

(How to verify changes)

- Navigate to monitor tab.
- Select all the filters. 
- Open networks tab, see the token call fired, observe the payload passed, it has selected resources/clusters' ids.

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

---

### CODE REVIEW

File: packages/manager/src/queries/cloudpulse/queries.ts [hunk 1]
```diff
@@ -84,6 +84,6 @@ export const queryFactory = createQueryKeys(key, {
 
   token: (serviceType: string | undefined, request: JWETokenPayLoad) => ({
     queryFn: () => getJWEToken(request, serviceType!),
-    queryKey: [serviceType],
+    queryKey: [serviceType, { resource_ids: request.resource_ids.sort() }],
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/cloudpulse/queries.ts:87]: Is the sort necessary? 

Could this be 
```suggestion
    queryKey: [serviceType, request],
```

or 

```suggestion
    queryKey: [serviceType, { resource_ids: request.resource_ids }],
```

Comment (ankitaakamai) [reply packages/manager/src/queries/cloudpulse/queries.ts:87]: Sorting is necessary for preventing unnecessary token creation for differently ordered same resource ids..

Comment (bnussman-akamai) [reply packages/manager/src/queries/cloudpulse/queries.ts:87]: Ah, I understand. Thanks 👍

**Comment by ankitaakamai** [comment]
@bnussman-akamai is there any additional feedback for this pr or it is good for approval?

**Review by bnussman-akamai — APPROVED** [review]

**Review by harsh-akamai — APPROVED** [review]
Changes look good! A small nit for the changeset

**Comment by ankitaakamai** [comment]
> Changes look good! A small nit for the changeset

@harsh-akamai done!

**Pull request merged**